system-tests/lib/cre/contracts/contracts.go

```diff
@@ -492,48 +495,72 @@ func DeployFeedsConsumer(testLogger zerolog.Logger, input *types.DeployFeedConsu
 		return nil, errors.Wrap(err, "input validation failed")
 	}
 
-	output, err := keystone_changeset.DeployFeedsConsumer(*input.CldEnv, &keystone_changeset.DeployFeedsConsumerRequest{
-		ChainSelector: input.ChainSelector,
-	})
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to deploy feeds_consumer contract")
+	deployConfig := df_changeset_types.DeployConfig{
+		ChainsToDeploy: []uint64{input.ChainSelector},
+		Labels:         []string{"data-feeds"},
+	}
+	dfOutput, dfErr := df_changeset.DeployCacheChangeset.Apply(*input.CldEnv, deployConfig)
+	if dfErr != nil {
+		return nil, errors.Wrap(dfErr, "failed to deploy data feed cache contract")
 	}
 
-	err = input.CldEnv.ExistingAddresses.Merge(output.AddressBook)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to merge address book")
+	mergeErr := input.CldEnv.ExistingAddresses.Merge(dfOutput.AddressBook)
+	if mergeErr != nil {
+		return nil, errors.Wrap(mergeErr, "failed to merge address book")
 	}
 
 	addresses, err := input.CldEnv.ExistingAddresses.AddressesForChain(input.ChainSelector)
 	if err != nil {
 		return nil, errors.Wrapf(err, "failed to get addresses for chain %d from the address book", input.ChainSelector)
 	}
 
-	var feedsConsumerAddress common.Address
+	var dataFeedsCacheAddress common.Address
 	for addrStr, tv := range addresses {
-		if strings.Contains(tv.String(), "FeedConsumer") {
-			feedsConsumerAddress = common.HexToAddress(addrStr)
-			testLogger.Info().Msgf("Deployed FeedConsumer contract at %s", feedsConsumerAddress.Hex())
+		if strings.Contains(tv.String(), "DataFeedsCache") {
+			dataFeedsCacheAddress = common.HexToAddress(addrStr)
+			testLogger.Info().Msgf("Deployed DataFeedsCache contract at %s", dataFeedsCacheAddress.Hex())
 			break
 		}
 	}
 
-	if feedsConsumerAddress == (common.Address{}) {
+	if dataFeedsCacheAddress == (common.Address{}) {
 		return nil, errors.New("failed to find FeedConsumer address in the address book")
 	}
 
-	out := &types.DeployFeedConsumerOutput{
-		FeedConsumerAddress: feedsConsumerAddress,
+	out := &types.DeployDataFeedsCacheOutput{
+		DataFeedsCacheAddress: dataFeedsCacheAddress,
 	}
-
 	input.Out = out
+
 	return out, nil
 }
 
-func ConfigureFeedsConsumer(testLogger zerolog.Logger, input *types.ConfigureFeedConsumerInput) (*types.ConfigureFeedConsumerOutput, error) {
+func hashTruncateName(name string) string {
```

Comment:
We need to copy it because we don't want to depend on these functions? https://github.com/smartcontractkit/chainlink/blob/95b2cba5412ec056567fb5750845b9167402875b/deployment/data-feeds/changeset/utils.go#L44